apps/rush-lib/src/api/EnvironmentConfiguration.ts

```diff
@@ -102,7 +108,9 @@ export class EnvironmentConfiguration {
         const normalizedEnvVarName: string = os.platform() === 'win32' ? envVarName.toUpperCase() : envVarName;
         switch (normalizedEnvVarName) {
           case EnvironmentVariableNames.RUSH_TEMP_FOLDER: {
-            EnvironmentConfiguration._rushTempFolderOverride = value;
+            EnvironmentConfiguration._rushTempFolderOverride = (value && !options.doNotNormalizePaths)
+              ? EnvironmentConfiguration._normalizeDeepestParentFolderPath(value) || value
```

Comment:
Isn't `EnvironmentConfiguration._normalizeDeepestParentFolderPath(value) || value` a boolean expression?

Reply:
JavaScript uses the `||` operator as a falsy (i.e. - `false`, `null`, `undefined`, and `''`) coalescing operator. So `false || true` evaluates to `true`, and `undefined || 'string'` evaluates to `string`.